test/test_xbuffer_adaptor.cpp

```diff
@@ -12,20 +12,21 @@
 namespace xt
 {
     using buffer_adaptor = xbuffer_adaptor<double*>;
+    using allocator = std::allocator<double>;
     using owner_adaptor = xbuffer_adaptor<double*&, acquire_ownership>;
 
     TEST(xbuffer_adaptor, owner_destructor)
     {
         size_t size = 100;
-        double* data = new double[size];
+        double* data = allocator{}.allocate(size);
```

Comment:
weird that we were not deleting in the tests.

Why using an allocator btw?

Reply:
we're not deleting because that's what the owning buffer adaptor is supposed to do and the allocator stuff because of otherwise mismathcing new / delete

Reply:
oh yes, that is right